### PR TITLE
Update test docker scripts to support sonar reporting

### PR DESCRIPTION
**Pull request by gpang**

### CODE REVIEW

**Comment by gpang** [comment]
Jenkins, test this please

File: dev/jenkins/run_docker.sh [hunk 1]
```diff
@@ -38,6 +38,13 @@ function main {
     run_args+=" -e ALLUXIO_GIT_CLEAN=true"
   fi
 
+  if [ -n "${ALLUXIO_SONAR_ARGS}" ]
+  then
+    # write out to a file, in case there are spaces in the args
+    echo "ALLUXIO_SONAR_ARGS=${ALLUXIO_SONAR_ARGS}" > /tmp/.env
```
Comment (ZacBlanco) [inline dev/jenkins/run_docker.sh]: you probably want this?

```
     echo "ALLUXIO_SONAR_ARGS=\"${ALLUXIO_SONAR_ARGS}\"" > /tmp/.env
```

File: dev/jenkins/build.sh [hunk 2]
```diff
@@ -30,3 +30,8 @@ then
   git clean -fdx
 fi
 mvn -Duser.home=/home/jenkins -T 4C clean install -Pdeveloper -Dmaven.javadoc.skip -Dsurefire.forkCount=${ALLUXIO_BUILD_FORKCOUNT} $@
+
+if [ -n "${ALLUXIO_SONAR_ARGS}" ]
```
Comment (ZacBlanco) [inline dev/jenkins/build.sh]: semicolon?
```suggestion
if [ -n "${ALLUXIO_SONAR_ARGS}" ];
```

File: dev/jenkins/run_docker.sh (hunk 1 above)
Comment (gpang) [reply dev/jenkins/run_docker.sh]: That doesn't seem to work. The quotes get included, which is not what I want. I think Docker already takes care if there are spaces in the values.

File: dev/jenkins/build.sh (hunk 2 above)
Comment (gpang) [reply dev/jenkins/build.sh]: I just copied whatever we do in this file and the other file, and none of them use the semicolon.

Comment (ZacBlanco) [reply dev/jenkins/build.sh]: It might be if the `fi` is on the same line

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**